app/org/sagebionetworks/bridge/validators/StudyConsentValidator.java

```diff
@@ -33,10 +47,22 @@ public void validate(Object object, Errors errors) {
             errors.rejectValue("documentContent", "is null or blank");
             return;
         }
+        // Assemble the final document and validate that it parses as XML. Otherwise the PDF generator
+        // will throw an error. We don't substitute any of the variables (these should not be XML, although
+        // we currently make no effort to prevent this).
         try {
-            InputStream stream = new ByteArrayInputStream(consent.getDocumentContent().getBytes(StandardCharsets.UTF_8));
-            DocumentBuilder parser = DocumentBuilderFactory.newInstance().newDocumentBuilder();
-            parser.parse(stream);
+            Map<String,String> map = Maps.newHashMap();
+            map.put("consent.body", consent.getDocumentContent());
+            String mergedDocument = BridgeUtils.resolveTemplate(consentBodyTemplate, map);
+            
+            SAXParserFactory factory = SAXParserFactory.newInstance();
+            factory.setValidating(false);
+            factory.setNamespaceAware(false);
+            SAXParser parser = factory.newSAXParser();
+            XMLReader reader = parser.getXMLReader();
+            
+            InputSource inputSource = new InputSource(new StringReader(mergedDocument));
+            reader.parse(inputSource);
```

Comment:
There's no close() method on the reader and I don't see this in examples. I don't think so.